src/System.Collections.NonGeneric/tests/Hashtable/AddTests.cs

```diff
@@ -0,0 +1,285 @@
+            sbl5.Append("key_m5"); // Backdoor duplicate!
+
+            sblWork1 = (StringBuilder)ht2[sbl5];
+            Assert.True(ht2.ContainsKey(sbl5));
+
+            ht2.Clear();
+          }
+
+        [Fact]
+        public void TestAddClearRepeatedly()
+        {
+            int inLoops0 = 2;
+            int inLoops1 = 2;
+
+            var ht2 = new Hashtable();
+            for (int aa = 0; aa < inLoops0; aa++)
+            {
+                for (int bb = 0; bb < inLoops1; bb++)
+                {
+                    ht2.Add("KEY: aa==" + aa + " ,bb==" + bb, "VALUE: aa==" + aa + " ,bb==" + bb);
+                }
+
+                Assert.Equal(inLoops1, ht2.Count);
+                ht2.Clear();
+            }
+        }
+
+        const int iterations = 1600;
+
+        [Fact]
+        public void TestDuplicatedKeysWithInitialCapacity()
+        {
+            // Make rehash get called because to many items with dupplicated keys have been added to the hashtable
+            var ht = new Hashtable(200);
+
+            for (int i = 0; i < iterations; i += 2)
+            {
+                ht.Add(new BadHashCode(i), i.ToString());
+                ht.Add(new BadHashCode(i + 1), (i + 1).ToString());
+
+                ht.Remove(new BadHashCode(i));
+                ht.Remove(new BadHashCode(i + 1));
+            }
+
+            for (int i = 0; i < iterations; i++)
+            {
+                ht.Add(i.ToString(), i);
+            }
+
+            for (int i = 0; i < iterations; i++)
+            {
+                Assert.Equal(i, (int)ht[i.ToString()]);
+            }
+        }
+
+        [Fact]
+        public void TestDuplicatedKeysWithDefaultCapacity()
+        {
+            // Make rehash get called because to many items with duplicated keys have been added to the hashtable
+            var ht = new Hashtable();
+
+            for (int i = 0; i < iterations; i += 2)
+            {
+                ht.Add(new BadHashCode(i), i.ToString());
+                ht.Add(new BadHashCode(i + 1), (i + 1).ToString());
+
+                ht.Remove(new BadHashCode(i));
+                ht.Remove(new BadHashCode(i + 1));
+            }
+
+            for (int i = 0; i < iterations; i++)
+            {
+                ht.Add(i.ToString(), i);
+            }
+
+            for (int i = 0; i < iterations; i++)
+            {
+                Assert.Equal(i, (int)ht[i.ToString()]);
+            }
+        }
+    }
+
+    public class BadHashCode
+    {
+        private uint _value;
+
+        public BadHashCode(int value)
+        {
+            this._value = (uint)value;
+        }
+
+        public override bool Equals(object o)
+        {
+            BadHashCode rhValue = o as BadHashCode;
+
+            if (null != rhValue)
+            {
+                return _value.Equals(rhValue);
+            }
+            else
+            {
+                throw new ArgumentException("o", "is not BadHashCode type actual" + o.GetType());
```

Comment:
In the event that it fired, this message would read better with a space after "actual"